app/src/main/java/org/lineageos/tv/launcher/utils/AppManager.kt

```diff
@@ -8,60 +8,46 @@ package org.lineageos.tv.launcher.utils
 import android.content.Context
 import android.content.Intent
 import android.net.Uri
-import androidx.core.content.edit
+import androidx.preference.PreferenceManager
+import org.lineageos.tv.launcher.ext.favoriteApps
 import org.lineageos.tv.launcher.model.AppInfo
-import org.lineageos.tv.launcher.model.Launchable
 
 object AppManager {
-    internal var onFavoriteAddedCallback: (packageName: String) -> Unit = {}
-    internal var onFavoriteRemovedCallback: (packageName: String) -> Unit = {}
-
-    fun getInstalledApps(context: Context): List<Launchable> {
-        val pm = context.packageManager
-        val intent = Intent(Intent.ACTION_MAIN, null).apply {
-            addCategory(Intent.CATEGORY_LAUNCHER)
-        }
-
-        return pm.queryIntentActivities(intent, 0).map {
-            AppInfo(it, context)
+    fun updateFavoriteApps(context: Context, installedApps: List<AppInfo>) {
+        val sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context)
+        val favoriteApps = sharedPreferences.favoriteApps.toMutableList()
+
+        // Remove apps from favorite if they got uninstalled
+        with(installedApps.map { it.packageName }) {
```

Comment:
Of all the changes, this is the one I struggled most to understand what gets called on what
Can you use named arguments in the lambdas via let?